src/StaticAnalyser.php

```diff
@@ -391,4 +389,25 @@ private function parseUseStatement(&$tokens, &$token, $parseContext)
 
         return $statements;
     }
+
+    private function extractTypeAndNextToken(array &$tokens, Context $parseContext): array
+    {
+        $type = UNDEFINED;
```

Comment:
When using the FQCN notation on non-primitive types like `\DateTimeInterface`, I think the character `\` will be in the tokens array as a character like `?`.

Reply:
oh snap, you're right. moreover, we will have to handle multi-segment namespaces too, e.g. `\App\Models\Foo` because that would tokenize as `T_NS_SEPARATOR` . `T_STRING` . `T_NS_SEPARATOR` . `T_STRING` . `T_NS_SEPARATOR` . `T_STRING`

We should probably abstract something out of `\OpenApi\StaticAnalyser::parseUseStatement` since much of that implementation logic can be reused here.

I will put this PR back into draft now and try to fix that up tonight.

Reply:
all right that should fix it 👍